File: 02-string-calculator/stringcalculator.py
import re

ERROR_MESSAGE = 'ERROR: The input cannot end with a delimiter'


def stringcalculator(numbers):
    if not len(numbers):
        # empty string
        return 0

    separator, numbers = re.split('//|\n', numbers)[1:]

    if numbers.endswith(separator):
        # the input ended with a delimiter
        raise Exception(ERROR_MESSAGE)

    split_numbers = numbers.split(separator)
    # return the sum
    return sum(map(int, split_numbers))

File: typescript/02-stringcalculator/stringcalculator.ts

export function StringCalculator(input: string): number {
  if (input.endsWith(',') || input.endsWith('\n')) {
    throw new Error("Error: the input can't end with a delimiter");
  }

  if (input === "") return 0;

  // we need filter to remove empty values
  var splitted = input.split(/\n/),
      sum = 0,
      delimiter = '',
      values = [];

  delimiter = splitted[0].replace('//', '');
  values = splitted[1].split(delimiter).filter(c => c);

  for (var i = 0; i < values.length; i++) {
    sum += +values[i];
  }

  return sum;
}
